pos_tare_generate_barcode/static/src/js/screens.js

```diff
@@ -0,0 +1,220 @@
+            // is the prefix, here 07, that is used to decide which type of
+            // barcode we're dealing with. A weight barcode has then two groups
+            // of five digits. The first group encodes the product id. Here the
+            // product id is 00000. The second group encodes the weight in
+            // grams. Here the weight is 12.345kg. The last digit of the barcode
+            // is a checksum, here symbolized by x.
+            var padding_size = 5;
+            var void_product_id = '0'.repeat(padding_size);
+            var weight_in_gram = weight * 1e3;
+
+            if (weight_in_gram >= Math.pow(10, padding_size)) {
+                throw new RangeError(_t("Maximum tare weight is 99.999kg"));
+            }
+
+            // Weight has to be padded with zeros.
+            var weight_with_padding = '0'.repeat(padding_size) + weight_in_gram;
+            var padded_weight = weight_with_padding.substr(
+                weight_with_padding.length - padding_size);
+            // Builds the barcode using a placeholder checksum.
+            var barcode = this.weight_barcode_prefix
+                .concat(void_product_id, padded_weight)
+                .concat(0);
+            // Compute checksum
+            var barcode_parser = this.pos.barcode_reader.barcode_parser;
+            var checksum = barcode_parser.ean_checksum(barcode);
+            // Replace checksum placeholder by the actual checksum.
+            return barcode.substr(0, 12).concat(checksum);
+        },
+        get_barcode_data: function () {
+            return this.barcode_data(this.get_tare_weight_in_kilogram());
+        },
+        lock_screen: function (locked) {
+            this._locked = locked;
+            if (locked) {
+                this.$('.print-label').addClass('disabled');
+            } else {
+                this.$('.print-label').removeClass('disabled');
+            }
+        },
+        print_web: function () {
+            window.print();
+            this.pos.get_order()._printed = true;
+        },
+        print: function () {
+            // See comment in print function of ReceiptScreenWidget
+            this.lock_screen(true);
+            var self = this;
+            setTimeout(function () {
+                self.lock_screen(false);
+            }, 1000);
+
+            this.print_web();
+            this.click_back();
+        },
+        click_back: function () {
+            this.close();
+            this.gui.show_screen(this.previous_screen);
+        },
+        renderElement: function () {
+            this._super();
+            var self = this;
+            this.$('.back').click(function () {
+                self.click_back();
+            });
+            this.$('.print-label').click(function () {
+                if (!self._locked) {
+                    self.print();
+                }
+            });
+        },
+        render_receipt: function () {
```

Comment:
a detail, but if you render `PosTareLabel`, you can name your function `render_label`